src/dataflow/src/source/postgres.rs

```diff
@@ -197,26 +197,47 @@ impl PostgresSourceReader {
             .parse()
             .or_else(|_| Err(anyhow!("invalid lsn"))));
         for info in publication_tables {
-            // TODO(petrosagg): use a COPY statement here for more efficient network transfer
-            let data = client
-                .simple_query(&format!(
-                    "SELECT * FROM {:?}.{:?}",
-                    info.namespace, info.name
-                ))
+            let relation_id: Datum = (info.rel_id as i32).into();
+            let reader = client
+                .copy_out_simple(
+                    format!(
+                        "COPY {:?}.{:?} TO STDOUT (FORMAT TEXT, DELIMITER '\t')",
+                        info.namespace, info.name
+                    )
+                    .as_str(),
+                )
                 .await?;
-            for msg in data {
-                if let SimpleQueryMessage::Row(row) = msg {
+
+            pin_mut!(reader);
+            while let Some(Ok(b)) = reader.next().await {
```

Comment:
This reader gives out `Bytes` instances that are not necessarily new line aligned so we need to handle the case where there is left over data from one message and stitch it together with the next message

Reply:
Do you know when this might happen? I've was going by https://www.postgresql.org/docs/14/sql-copy.html#id-1.9.3.55.9.2 which says `When the text format is used, the data read or written is a text file with one line per table row. ... COPY TO will terminate each row with a Unix-style newline (“\n”).` / the source here shows (I think) postgres flushes after each tuple is written https://github.com/postgres/postgres/blob/master/src/backend/commands/copyto.c#L883-L900 -- maybe if a tuple exceeds the size of the send buffer?

Reply:
Exceeding the send buffer is one, or maybe if there is a TCP proxy between materialize and postgres it could disassemble and reassemble TCP frames at arbitrary points, etc.

Reply:
Hm since this is all TCP I'm not sure follow -- if postgres is always sending back each row/tuple as an individual CopyData message, doesn't that need to reassemble correctly on the client? Otherwise it seems the tags and message reading would get mangled

We have an existing test for a 16MB value (send buffer is 8KB) which works correctly, and I have a local version that makes the test a little more challenging which also works.

Reply:
Handy links I found along the way:

* https://github.com/postgres/postgres/blob/master/src/backend/commands/copyto.c#L248-L249 -- each tuple is sent as an individual CopyData message
* https://www.postgresql.org/docs/14/libpq-copy.html#LIBPQ-COPY-RECEIVE docs on the C library